crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -69,6 +72,112 @@ struct CheckpointStoreTables {
 
     /// Stores certified checkpoints
     certified_checkpoints: DBMap<CheckpointSequenceNumber, CertifiedCheckpointSummary>,
+    /// Map from checkpoint digest to certified checkpoint
+    checkpoint_by_digest: DBMap<CheckpointDigest, CertifiedCheckpointSummary>,
+
+    /// Watermarks used to determine the highest verified and fully synced checkpoints
+    watermarks: DBMap<CheckpointWatermark, (CheckpointSequenceNumber, CheckpointDigest)>,
+}
+
+impl CheckpointStoreTables {
+    pub fn get_checkpoint_by_digest(
+        &self,
+        digest: &CheckpointDigest,
+    ) -> Result<Option<VerifiedCheckpoint>, TypedStoreError> {
+        self.checkpoint_by_digest
+            .get(digest)
+            .map(|maybe_checkpoint| maybe_checkpoint.map(VerifiedCheckpoint::new_unchecked))
+    }
+
+    pub fn get_checkpoint_by_sequence_number(
+        &self,
+        sequence_number: CheckpointSequenceNumber,
+    ) -> Result<Option<VerifiedCheckpoint>, TypedStoreError> {
+        self.certified_checkpoints
+            .get(&sequence_number)
+            .map(|maybe_checkpoint| maybe_checkpoint.map(VerifiedCheckpoint::new_unchecked))
+    }
+
+    pub fn get_highest_verified_checkpoint(
+        &self,
+    ) -> Result<Option<VerifiedCheckpoint>, TypedStoreError> {
+        let highest_verified = if let Some(highest_verified) =
+            self.watermarks.get(&CheckpointWatermark::HighestVerified)?
+        {
+            highest_verified
+        } else {
+            return Ok(None);
+        };
+        self.get_checkpoint_by_digest(&highest_verified.1)
+    }
+
+    pub fn get_highest_synced_checkpoint(
+        &self,
+    ) -> Result<Option<VerifiedCheckpoint>, TypedStoreError> {
+        let highest_synced = if let Some(highest_synced) =
+            self.watermarks.get(&CheckpointWatermark::HighestSynced)?
+        {
+            highest_synced
+        } else {
+            return Ok(None);
+        };
+        self.get_checkpoint_by_digest(&highest_synced.1)
+    }
+
+    pub fn get_checkpoint_contents(
+        &self,
+        digest: &CheckpointContentsDigest,
+    ) -> Result<Option<CheckpointContents>, TypedStoreError> {
+        self.checkpoint_content.get(digest)
+    }
+
+    pub fn insert_verified_checkpoint(
+        &self,
+        checkpoint: VerifiedCheckpoint,
+    ) -> Result<(), TypedStoreError> {
+        self.certified_checkpoints
```

Comment:
Same as with other place - perhaps it's best to cover those two updates in the same batch.
Also given that we have other place that do same thing, perhaps it's even possible to refactor both updates in some method (e.g. something like `insert_certified_checkpoint(CertifiedCheckpointSummary)` that updates both `certified_checkpoints` and `checkpoint_by_digest `)